### PR TITLE
Show fully qualified resources name

### LINKED ISSUE #322: resources displayed in OUTPUT column in workload get  `supply chain` and `delivery` sections need to be disambiguated
### Please fill out the issue checklist below and provide ALL the requested information.

- [X] I reviewed open and closed Github issues that may be related to my problem.
- [X] I am reporting a bug that others will be able to reproduce.

### Describe the bug
We provided a reference to the stamped out resource associated with a given stage of the supply chain or delivery steps for a workload.

The intention of this was to provide an easy copy/paste flow a use could execute when troubleshooting.
The current format is `kind/name`.
For example, if the build failed, they could copy/paste the stamped out resource from the supply-chain OUTPUT column and `kubectl describe` it.

There were some changes to underlying apis which broke the intended troubleshooting path.
We learned that the only way to definitively reference the stamped out resource is to provide the fully qualified resource name rather than `Kind`

### Expected behavior
For example, when I copy past the `PodIntent/spring-sensors-bogus-consumer-web` into `kubectl describe`, I expect the kubeAPI to return the full description of that resource.

what actually happens today is:
```
kubectl describe PodIntent/spring-sensors-bogus-consumer-web
Warning: conventions.apps.tanzu.vmware.com/v1alpha1 PodIntent is deprecated; use conventions.carto.run/v1alpha1 PodIntent instead
Error from server (NotFound): podintents.conventions.apps.tanzu.vmware.com "spring-sensors-bogus-consumer-web" not found
```
### Steps to Reproduce
1. create a workload 
2. once it's running, run `tanzu apps workload get APPNAME`
3. copy the name of the stamped our resource in the OUTPUT column for the `config-provider` row of the table
4. kubectl describe it (e.g. `kubectl describe PodIntent/spring-sensors-bogus-consumer-web`)
5. notice the 'warning' message and note there's no describe output returned. 

### Proposed Solution - applies to both `supply chain` and `delivery` sections:
1. [ ] Change column headings from `RESOURCE    READY   HEALTHY   TIME   OUTPUT` ==>  `NAME    READY   HEALTHY   UPDATED   RESOURCE`
2. [ ] update the `RESOURCE column value to display the fully qualified resource ref rather than the `kind`
3. [ ] update command reference docs if/where the old output is displayed (eg. if there is example output)
4. [ ] provide a release note

example:
```
...
NAME               READY   HEALTHY   UPDATED    RESOURCE
source-provider    True    True      5d18h      gitrepositories.source.toolkit.fluxcd.io/spring-sensors-bogus-consumer-web
image-provider     True    True      44h        images.kpack.io/spring-sensors-bogus-consumer-web
config-provider    True    True      6d23h      podintents.conventions.carto.run/spring-sensors-bogus-consumer-web
...
```

### PR DESCRIPTION
**Pull request by warango4**
Pull request

### What this PR does / why we need it
In existing `resource` column (formerly `output`) in `workload get` resources table, now it's being displayed the fully qualified resource name so user can check it with commands such as `kubectl describe`.

### Which issue(s) this PR fixes

Fixes #322 

### Describe testing done for PR

- Updated unit testing

### Additional information or special notes for your reviewer

If user has an older Cartographer version on their cluster, the `stampedref.resource` field won't be available, so output will keep showing `stampedref.kind/stampedref.name`

Signed-off-by: Wendy Arango <warango@vmware.com>

### CODE REVIEW

**Comment by warango4** [comment]
image

**Comment by warango4** [comment]
With `carto v0.5.3`, this is the output

image

**Review by odinnordico — DISMISSED** [review]

**Review by anibmurthy — DISMISSED** [review]
lgtm

File: pkg/apis/cartographer/v1alpha1/common.go [hunk 1]
```diff
@@ -124,3 +124,11 @@ type OwnerStatus struct {
 	// https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties
 	Conditions []metav1.Condition `json:"conditions,omitempty"`
 }
+
+type StampedRef struct {
+	*corev1.ObjectReference `json:",inline,omitempty"`
```
[the remark this answers is not in the document]
Comment (warango4) [reply pkg/apis/cartographer/v1alpha1/common.go:129]: It's not the variable name. This is meant to encode this field as a JSON object

File: pkg/apis/cartographer/v1alpha1/common.go [hunk 2]
```diff
@@ -78,7 +78,7 @@ type RealizedResource struct {
 	Name string `json:"name"`
 
 	// StampedRef is a reference to the object that was created by the resource
```
[the remark this answers is not in the document]
Comment (warango4) [reply pkg/apis/cartographer/v1alpha1/common.go:80]: https://github.com/vmware-tanzu/cartographer/blob/main/pkg/apis/v1alpha1/common.go#L219-L220

I'm following Carto on this

File: pkg/apis/cartographer/v1alpha1/zz_generated.deepcopy.go [hunk 3]
```diff
@@ -471,6 +471,26 @@ func (in *Source) DeepCopy() *Source {
 	return out
 }
 
+// DeepCopyInto is an autogenerated deepcopy function, copying the receiver, writing into out. in must be non-nil.
+func (in *StampedRef) DeepCopyInto(out *StampedRef) {
+	*out = *in
```
[the remark this answers is not in the document]
Comment (warango4) [reply pkg/apis/cartographer/v1alpha1/zz_generated.deepcopy.go:476]: this is an autogenerated file. Even though, I don't think it controls it. While line [383](https://github.com/vmware-tanzu/apps-cli-plugin/blob/e8fe31486323ed2fe7766ad33bb10f4307db67d4/pkg/apis/cartographer/v1alpha1/zz_generated.deepcopy.go#L383-L387) is asking if type `StampedRef` is empty, line [477](https://github.com/vmware-tanzu/apps-cli-plugin/blob/e8fe31486323ed2fe7766ad33bb10f4307db67d4/pkg/apis/cartographer/v1alpha1/zz_generated.deepcopy.go#L477-L481) is asking if type `ObjectReference` is empty

File: pkg/apis/cartographer/v1alpha1/common.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (shaheerkootteeri) [reply pkg/apis/cartographer/v1alpha1/common.go:129]: I was checking if we can assign a name to it, as we had a name stamped ref for this type

File: pkg/printer/workload_status_printer.go [hunk 4]
```diff
@@ -167,7 +167,9 @@ func findConditionReady(conditions []metav1.Condition, strReadyCondition string)
 func getOutputRef(resource *cartov1alpha1.RealizedResource) string {
 	ref := printer.Sfaintf("not found")
 	if resource != nil && resource.StampedRef != nil {
-		if resource.StampedRef.Kind != "" || resource.StampedRef.Name != "" {
+		if resource.StampedRef.Resource != "" && resource.StampedRef.Name != "" {
+			ref = fmt.Sprintf("%s%s%s", resource.StampedRef.Resource, "/", resource.StampedRef.Name)
```
Comment (odinnordico) [inline pkg/printer/workload_status_printer.go:171]:
```suggestion
			ref = fmt.Sprintf("%s/%s", resource.StampedRef.Resource, resource.StampedRef.Name)
```

File: pkg/printer/workload_status_printer.go [hunk 5]
```diff
@@ -167,7 +167,9 @@ func findConditionReady(conditions []metav1.Condition, strReadyCondition string)
...
+		} else if resource.StampedRef.Kind != "" && resource.StampedRef.Name != "" {
 			ref = fmt.Sprintf("%s%s%s", resource.StampedRef.Kind, "/", resource.StampedRef.Name)
```
Comment (odinnordico) [inline pkg/printer/workload_status_printer.go:173]:
```suggestion
			ref = fmt.Sprintf("%s%s", resource.StampedRef.Kind, resource.StampedRef.Name)
```

**Review by shaheerkootteeri — APPROVED** [review]
lgtm

**Pull request merged**